spec/requests/api/v1/provider_spec.rb

```diff
@@ -213,15 +218,78 @@
               headers: { 'HTTP_AUTHORIZATION' => credentials },
               params: { changed_since: 10.minutes.ago.utc.iso8601 }
 
-          json = JSON.parse(response.body)
-          returned_provider_codes = json.map { |provider| provider["institution_code"] }
+          returned_provider_codes = get_provider_codes_from_body(response.body)
 
           expect(returned_provider_codes).not_to include old_provider.provider_code
           expect(returned_provider_codes).to include updated_provider.provider_code
           expect(returned_provider_codes).to include provider_with_updated_enrichment.provider_code
           expect(returned_provider_codes).to include provider_with_updated_site.provider_code
         end
       end
+
+      it 'includes correct next link in response headers' do
+        create(:provider, provider_code: "LAST1", age: 10.minutes.ago)
+
+        timestamp_of_last_provider = 2.minutes.ago
+        last_provider_in_results = create(:provider, provider_code: "LAST2", age: timestamp_of_last_provider)
+
+        get '/api/v1/providers',
+          headers: { 'HTTP_AUTHORIZATION' => credentials },
+          params: { changed_since: 30.minutes.ago.utc.iso8601 }
+
+        expect(response.headers).to have_key "Link"
+        expected = /#{request.base_url + request.path}\?changed_since=#{(timestamp_of_last_provider + 1.second).utc.iso8601}&from_provider_id=#{last_provider_in_results.id}; rel="next"$/
+        expect(response.headers["Link"]).to match expected
+      end
+
+      it 'includes correct next link when there is an empty set' do
+        provided_timestamp = 5.minutes.ago.utc.iso8601
+
+        get '/api/v1/providers',
+          headers: { 'HTTP_AUTHORIZATION' => credentials },
+          params: { changed_since: provided_timestamp }
+
+        expected = /#{request.base_url + request.path}\?changed_since=#{provided_timestamp}&from_provider_id=; rel="next"$/
+        expect(response.headers["Link"]).to match expected
+      end
+
+      context "with many providers" do
+        before do
+          101.times do |i|
+            create(:provider, provider_code: "PROV#{i + 1}", age: (200 - i).minutes.ago)
```

Comment:
I would expect creating 101 providers would be quite slow ... I would suggest that we either:

1. set a `per_page` option to something smaller, like 10, and then rely on other tests to test that the default `per_page` is set to 100
2. if that feels too artificial, then at least stop the provider from creating associated data like `courses`, `sites` and `enrichments`.